src/v/archival/types.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "seastar/core/sstring.hh"
+#include "seastarx.h"
+#include "utils/named_type.h"
+
+namespace archival {
+
+/// Segment file name without working directory,
+/// expected format: <base-offset>-<term-id>-<revision>.log
+using segment_name = named_type<ss::sstring, struct archival_segment_name_t>;
+/// Segment path in S3, expected format:
+/// <prefix>/<ns>/<topic>/<part-id>_<rev>/<base-offset>-<term-id>-<revision>.log
+using remote_segment_path
```

Comment:
nit: should these paths be `std::filesystem::path` ? string is ok. how expensive are paths? same byte size no?

Reply:
they can still be `named_type<std::filesystem::path, struct archival_segment_name_t>` for example